### PR TITLE
Sync polymer2 to current master

### PR DESCRIPTION
**Pull request by stephanwlee**
Conflicts manually resolved for 
- third_party/js.bzl 
- third_party/polymer.bzl
Most changes were `https://mirror.bazel.build` -> `http://mirror.tensorflow.org`.
- Unbreaking Polymer from Closure Compiler changes.
  - https://github.com/google/closure-compiler/issues/3240
  - JScomp changes behavior of Polymer: https://gist.github.com/stephanwlee/493b010cc77f57d193cf9c1abd7dc763

### CODE REVIEW

**Comment by wchargin** [comment]
> Most changes were `https://mirror.bazel.build` -> `http://mirror.tensorflow.org`.

I’ve updated the mirror for these new links, per <http://b/133880558>.

File: tensorboard/components/tensorboard.html [hunk 1]
```diff
@@ -22,7 +22,14 @@
...
 <link rel="import" href="analytics.html">
 <script src="web-animations-js/web-animations-next-lite.min.js"></script>
-<script src="webcomponentsjs/webcomponents-lite.js"></script>
+<script jscomp-minify src="webcomponentsjs/webcomponents-lite.js"></script>
+<!-- HACK: Polymer2 (not 3) makes use of `super` in static methods which gets
```
Comment (stephanwlee) [inline tensorboard/components/tensorboard.html]: This is an important changes.

**Comment by stephanwlee** [comment]
Will manually flip the CLA after LGTM.

File: tensorboard/components/tensorboard.html [hunk 2]
```diff
@@ -22,7 +22,14 @@
...
 <link rel="import" href="analytics.html">
 <script src="web-animations-js/web-animations-next-lite.min.js"></script>
-<script src="webcomponentsjs/webcomponents-lite.js"></script>
+<script jscomp-minify src="webcomponentsjs/webcomponents-lite.js"></script>
```
Comment (wchargin) [inline tensorboard/components/tensorboard.html]: Is the new `jscomp-minify` directive semantic or just nice to have?

File: tensorboard/components/tensorboard.html (hunk 1 above)
Comment (wchargin) [reply tensorboard/components/tensorboard.html]: Ack; thanks for calling this out. Also related:
<https://github.com/polymer/polymer/commit/cc7702b476e3b695eb0e238463fb65d7b1315046>

**Comment by wchargin** [comment]
Overriding CLAs because all new commits are already in master:

```
$ git rev-parse HEAD origin/master
1b9863777c0c54a72ac4f202fbd07a255c596ebf
cde14ef9e2a56a508a7fdaf740915bdf07300f14
$ git rev-list HEAD ^HEAD^1 ^origin/master
1b9863777c0c54a72ac4f202fbd07a255c596ebf
$ git show --no-patch --format=%s 1b9863777c0c54a72ac4f202fbd07a255c596ebf
Merge remote-tracking branch 'upstream/master' into polymer2
```

**Comment by wchargin** [comment]
Friendly reminder that you wanted to merge this via a merge commit. Feel
free to temporarily enable the “Allow merge commits” option here:
<https://github.com/tensorflow/tensorboard/settings#merge-button-settings>

File: tensorboard/components/tensorboard.html (hunk 2 above)
Comment (stephanwlee) [reply tensorboard/components/tensorboard.html]: Not a nice-to-have. I actually looked into this a little bit more carefully (specifically followed the difference between two code paths here: https://github.com/tensorflow/tensorboard/blob/ff05d1af8d118d0add237766980d31163a94c40b/tensorboard/java/org/tensorflow/tensorboard/vulcanize/Vulcanize.java#L324-L347) and found out that webcomponents-lite's `@nocompile` annotation causes JSCompiler to not include the source in the compiled JavaScript (both source and webpath references are missing). I would want to learn why JSCompiler decided to omit the source completely with the annotation but I decided to stop further investigation for now.

**Pull request merged**